lib/livebook_web/helpers.ex

```diff
@@ -76,6 +76,9 @@ defmodule LivebookWeb.Helpers do
     )
     |> Phoenix.HTML.safe_to_string()
     |> String.split("\n")
+    # Keep trailing newline character so that a
+    # copy/paste of the markup retains newlines.
+    |> Enum.map(&[&1, "\n"])
     |> Enum.map(&Phoenix.HTML.raw/1)
   end
```

Comment:
This is a more efficient version that does not remove the newline only to add it back. It also does less passes on the data:

```suggestion
    |> newlines_to_raw()
  end

  defp newlines_to_raw(bin) do
    {parts, last} =
      bin
      |> :binary.matches("\n")
      |> Enum.reduce({[], 0}, fn {pos, _}, {parts, previous} ->
        {[raw_part(bin, previous, pos + 1) | parts], pos + 1}
      end)

    if last == byte_size(bin) do
      Enum.reverse(parts)
    else
      Enum.reverse([raw_part(bin, last, byte_size(bin)) | parts])
    end
  end

  defp raw_part(bin, start, stop) do
    bin
    |> binary_part(start, stop - start)
    |> Phoenix.HTML.raw()
  end
```

I haven't tested it though!

Reply:
Given @jonatanklosko's comment though, you can safely ignore this one. :)